composer/profiler/profiler.py

```diff
@@ -143,6 +143,23 @@ def __init__(
                                profile_net=sys_prof_net,
                                stats_thread_interval_seconds=sys_prof_stats_thread_interval_seconds))
 
+        if torch_prof_memory_filename is not None:
+            if not (torch_prof_with_stack and torch_prof_record_shapes and torch_prof_profile_memory):
+                raise ValueError(
+                    f'torch_prof_memory_filename is set. Generating the memory timeline graph requires all the three flags torch_prof_with_stack, torch_prof_record_shapes, and torch_prof_profile_memory to be true. Got torch_prof_with_stack={torch_prof_with_stack}, torch_prof_record_shapes={torch_prof_record_shapes}, torch_prof_profile_memory={torch_prof_profile_memory}'
+                )
+            log.info(
+                f'Memory profiling is enabled and uses {torch_prof_memory_filename} as the filename to generate the memory timeline graph'
+            )
+
+        if (torch_prof_with_stack and torch_prof_record_shapes and
+                torch_prof_profile_memory) and torch_prof_memory_filename is None:
+
+            log.info(
+                f'Memory profiling is enabled but torch_prof_memory_filename is not set. Using {"rank{rank}.{batch}.pt.memory_trace.html"} as the filename for the memory timeline graph.'
+            )
+            torch_prof_memory_filename = 'rank{rank}.{batch}.pt.memory_trace.html'
+
```

Comment:
Can we remove this? I think we shouldn't generate the memory timeline graph by default if it's not specified. `torch_prof_profile_memory`

Reply:
done.